build.gradle

```diff
@@ -79,6 +81,9 @@ subprojects {
     tasks.withType(JavaCompile) {
       // Treat each compiler warning (esp. the ones coming from Checker Framework) as an error.
       options.compilerArgs << '-Werror'
+      if (System.getenv('CI') != 'true') {
+        options.forkOptions.jvmArgs += ['-Xmx12g', '-XX:+HeapDumpOnOutOfMemoryError']
```

Comment:
Skipping `-XX:MaxPermSize` - deprecated since Java 8

Reply:
IMHO this (local options) should be mentioned in the contribution guidelines anyway

Reply:
@mkondratek I reworked that slightly (also decreased the default limit, 12GB was kinda a looot), pls see the updated version

Reply:
this is fine :fire: